### ISSUE TITLE
Game freeze when try to generate structure in void

### ISSUE THREAD

**Issue by SettingDust** [TeamChocoQuest/ChocolateQuestRepoured#324]
**General Information**
To check a checkbox: Insert a X between the square brackets, you need to remove the space though.
 - I play...
   - [ ] With a large modpack
   - [x] Only with CQR and it's dependencies
 - The issue occurs in...
   - [x] Singleplayer
   - [x] Multiplayer
 - [x] I have searched for this or a similar issue before reporting and it was either (1) not previously reported, or (2) previously fixed and I'm having the same problem.
 - [x] I am using the latest version of the mod (all versions can be found on github under releases)
 - [x] I read through the FAQ and I could not find something helpful ([FAQ](https://wiki.cq-repoured.net/index.php?title=FAQ))
 - [x] I reproduced the bug without any other mod's except forge, cqr and it's dependencies
 - [ ] The game crashes because of this bug
 
**Describe the bug**
Game freeze when try to generate structure in void. I want to generate a world with only one dungeon.
Tested with higher Y(/setblock ~ 63 ~ dirt). 

**To Reproduce**
1. Create a flat world with void preset
2. try to use structure placer beside test
 
**Versions (Note that 'latest' is NOT a valid version!**
Chocolate Quest Repoured: 1.12.2-2.6.6B
Forge: v14.23.5.2860

**Log File**
Just stuck after `[16:02:07] [Server thread/INFO] (cqrepoured) Generating dungeon RandomizedCastle at BlockPos{x=10, y=4, z=11}`

**Comment by DerToaster98** [comment]
Could you try with a different dungeon type?
The log just says it tried to generate a randomized castle. 
Interesting for us would be if that is a problem with just that type or in general.

**Comment by SettingDust** [comment]
> Could you try with a different dungeon type? The log just says it tried to generate a randomized castle. Interesting for us would be if that is a problem with just that type or in general.

Tried. Same result.
```
[16:14:03] [Server thread/INFO] (cqrepoured) Generated dungeon Test at BlockPos{x=12, y=64, z=10}
[16:14:06] [Server thread/INFO] (cqrepoured) Generating dungeon VegetatedCave at BlockPos{x=12, y=65, z=10}
[16:14:11] [Server thread/INFO] (cqrepoured) Generated dungeon VegetatedCave at BlockPos{x=12, y=65, z=10}
[16:14:11] [Server thread/WARN] (MinecraftServer) Can't keep up! Did the system time change, or is the server overloaded? Running 4969ms behind, skipping 99 tick(s)
[16:14:19] [Server thread/INFO] (cqrepoured) Generating dungeon StrongholdLinearSnow at BlockPos{x=11, y=63, z=12}
```
In this test, Stronghold is freeze. Still testing for more dungeon.

**Comment by DerToaster98** [comment]
ok, then it is a general problem. No need to try more.

An important information for us would be if you deleted the dungeon configs before applying the update

**Comment by SettingDust** [comment]
> ok, then it is a general problem. No need to try more.
> 
> An important information for us would be if you deleted the dungeon configs before applying the update

It's a new instance

**Comment by DerToaster98** [comment]
ok.

Then try disabling the support hill for some dungeon that has it enabled and then restart the game and try generating that dungeon.

I assume the problem is the support hill generation

**Comment by SettingDust** [comment]
> 

Spend some time for find the option. It's ok now!
StrongholdLinearSnow
```
buildsupportplatform = false
supportblock = minecraft:stone
supportblocktop = minecraft:snow
```

**Comment by DerToaster98** [comment]
Then i‘ll fix it later. Already found the problematic spot

**Comment by SettingDust** [comment]
> Then i‘ll fix it later. Already found the problematic spot

But the part below stronghold are disappear

**Comment by DerToaster98** [comment]
That is not really exact and i don't know what you mean.

**Comment by SettingDust** [comment]
> That is not really exact and i don't know what you mean.

stronghold lower part are disappeared
![图片](https://user-images.githubusercontent.com/19922286/149902683-53fd2ac6-c197-4d74-8fc7-cde3162fe439.png)

**Comment by DerToaster98** [comment]
If that is at y = 0 that is completely normal and supposed. The stronghold needs a certain amount of space underneath the surfaceto actually generate the rooms.

**Comment by SettingDust** [comment]
> 

Alright. Thanks a lot.

**Comment by DerToaster98** [comment]
should be patched in v2.6.7B
